pkg/reconciler/openshift/tektonhub/extension.go

```diff
@@ -0,0 +1,173 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tektonhub
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/go-logr/zapr"
+	mfc "github.com/manifestival/client-go-client"
+	mf "github.com/manifestival/manifestival"
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/client-go/route/clientset/versioned/scheme"
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	"github.com/tektoncd/operator/pkg/client/clientset/versioned"
+	operatorclient "github.com/tektoncd/operator/pkg/client/injection/client"
+	"github.com/tektoncd/operator/pkg/reconciler/common"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/injection"
+	"knative.dev/pkg/logging"
+)
+
+const (
+	apiDirName    string = "api"
+	apiSecretName string = "api"
+	apiRouteName  string = "api"
+	authRouteName string = "auth"
+)
+
+func OpenShiftExtension(ctx context.Context) common.Extension {
+	logger := logging.FromContext(ctx)
+	mfclient, err := mfc.NewClient(injection.GetConfig(ctx))
+	if err != nil {
+		logger.Fatalw("error creating client from injected config", zap.Error(err))
+	}
+	mflogger := zapr.NewLogger(logger.Named("manifestival").Desugar())
+	manifest, err := mf.ManifestFrom(mf.Slice{}, mf.UseClient(mfclient), mf.UseLogger(mflogger))
+	if err != nil {
+		logger.Fatalw("error creating initial manifest", zap.Error(err))
+	}
+
+	ext := openshiftExtension{
+		operatorClientSet: operatorclient.Get(ctx),
+		kubeClientSet:     kubeclient.Get(ctx),
+		manifest:          manifest,
+	}
+	return ext
+}
+
+type openshiftExtension struct {
+	operatorClientSet versioned.Interface
+	kubeClientSet     kubernetes.Interface
+	manifest          mf.Manifest
+}
+
+func (oe openshiftExtension) Transformers(comp v1alpha1.TektonComponent) []mf.Transformer {
+	return nil
+}
+
+func (oe openshiftExtension) PreReconcile(ctx context.Context, tc v1alpha1.TektonComponent) error {
+	th := tc.(*v1alpha1.TektonHub)
+	logger := logging.FromContext(ctx)
+	koDataDir := os.Getenv(common.KoEnvKey)
+	targetNs := th.Spec.GetTargetNamespace()
+	hubDir := filepath.Join(koDataDir, "tekton-hub", common.TargetVersion(th), apiDirName)
+	manifest := oe.manifest.Append()
+
+	if err := common.AppendManifest(&manifest, hubDir); err != nil {
+		return err
+	}
+	manifest, err := manifest.Transform(
+		mf.InjectOwner(th),
+		mf.InjectNamespace(targetNs),
+	)
+	if err != nil {
+		logger.Error("failed to transform manifest")
+		return err
+	}
+
+	// Just apply the routes and nothing else
+	if err := manifest.Filter(mf.ByKind("Route")).Apply(); err != nil {
+		return err
+	}
```

Comment:
replace this with installerSet creation.
1. check for existing installerSet of this type (eg: 'hub-routes') 
1. if there are more than 1 delete it reconcile again
1. create/update instaler set
1. create and installerSet with createdBy, version, type labels.
1. also in extension finalizer add logic to query and delete the installerSets created in this extension.

eg: 
- https://github.com/tektoncd/operator/blob/a0cd67e5c428babf1f76316a14cf9c4526055cab/pkg/reconciler/openshift/tektonpipeline/extension.go#L125
- https://github.com/tektoncd/operator/blob/a0cd67e5c428babf1f76316a14cf9c4526055cab/pkg/reconciler/openshift/tektonpipeline/extension.go#L204